test/layers/conv.jl

```diff
@@ -20,6 +20,9 @@ adj = [0. 1. 0. 1.;
 
             Y = gc(X)
             @test size(Y) == (out_channel, N)
+            # Test that the gradient can be computed
+            Zygote.gradient(() -> sum(gc(X)), params(gc))
```

Comment:
To check the gradient, you could reference [gradcheck](https://github.com/FluxML/Zygote.jl/blob/master/test/gradcheck.jl) here.
I think the gradient test can be isolated as an another file `test/layer/grad.jl`.

Reply:
Oh I didn't know the gradtest function, very useful thank you!

Reply:
It seems to be in their `test` directory, therefore not available in the module, do you think we should copy the functions?

Reply:
And unfortunately the gradcheck function only works with arrays as parameters. However, our params are of Flux' type `Params` and don't seem to work with that function. Furthermore, I think using ngradient on (kinda) complicated layers like the messagepassing ones would take too much time.

Reply:
I think gradcheck can be modified slightly to handle `Params`. It's not that hard though. Could you do this?

Reply:
I'd love to but I'd say I'm not sure how I would do it. Could you point me in the right direction please?